### ISSUE TITLE
Not all base quantities for Floors/IfcSlab in Rvt23

### ISSUE THREAD

**Issue by louistrue** [Autodesk/revit-ifc#515]
In Revit 23 with latest exporter version, base quantities get only exported to IFC4 (Reference view, structural) for some slabs. Others get no base quantities...

**Comment by parrela** [comment]
I don't see this myself - could you attach a model that shows the problem? Thanks, I will check it out and file an issue.

**Comment by louistrue** [comment]
NetWeight, NetArea and GrossWeight is missing on all slabs
Width and Length only exported for some slabs
GrossVolume, GrossArea and Depth fails to export for a few as well

![Snag_304bd2a](https://user-images.githubusercontent.com/78563314/195502061-582a04f9-2ab3-439f-a46a-8fc260887c92.png)
![Snag_304bd59](https://user-images.githubusercontent.com/78563314/195502064-3dfa5042-1300-4115-bdee-0cc663882792.png)
![Snag_304bd78](https://user-images.githubusercontent.com/78563314/195502066-70006667-c28f-4aba-bec1-9e05aa65f269.png)

[Rvt23_Floor_BaseQ.zip](https://github.com/Autodesk/revit-ifc/files/9770816/Rvt23_Floor_BaseQ.zip)

**Comment by louistrue** [comment]
@parrela can you provide an email to send it? We dont upload Rvt to Git...

**Comment by parrela** [comment]
andrew.parrella@autodesk.com

**Comment by Moult** [comment]
I've also been experiencing missing GrossVolumes on various concrete beams (the beams are also exported as advanced BReps for some reason).

**Comment by AngelVelezSosa** [comment]
The beams are exported as advancedbreps in ifc2x3?

**Comment by Moult** [comment]
@AngelVelezSosa no, in IFC4. The advanced brep problem is already captured in #480.

**Comment by AngelVelezSosa** [comment]
Ok, so "for some reason" just means "I thought they'd be an extrusion"?

**Comment by Moult** [comment]
Yep.

**Comment by ArtemHnatko** [comment]
### Update: Not all base quantities for Floors/IfcSlab in Rvt23 (#515)

Hello @louistrue,

Thank you for reporting this issue and providing the sample files, which were very helpful in our investigation.

We've reviewed this behavior and gathered additional context from related discussions and internal analysis. Here's a summary of our findings regarding base quantities for `IfcSlab` (and similar elements):

**1. `NetWeight` and `GrossWeight`:**
* These quantities are **not natively calculated or stored** as standard element properties within Revit.
* Therefore, their absence in IFC exports is **currently "by design"** due as a limitation of Revit's internal architecture regarding weight information.
* **Workaround:** To export `NetWeight` and `GrossWeight`, users must create dedicated **Project Parameters** (preferably Shared Parameters) in Revit and populate their values, often via formulas in schedules, as Revit does not provide direct element properties for these.

**2. Other Quantities (`Width`, `Length`, `GrossVolume`, `GrossArea`, `Depth`, `NetArea`, `NetVolume`, `Perimeter`):**
* **`GrossArea`, `GrossVolume`, `NetArea`, `NetVolume`, and `Perimeter`** are generally expected to export correctly.
* **`Width` and `Length`** are primarily exported for **rectangular slabs**. Their behavior for non-rectangular or complex slabs may vary.
* **`Depth` and `Length`** (specifically for slabs) are **supported for IFC4 Reference View exports**, but are **not supported for IFC2x3 exports**. This aligns with the IFC schema definitions for these quantities in different versions.

**Conclusion:**
The behavior observed for `NetWeight` and `GrossWeight` is a known limitation requiring user-defined parameters. For other quantities, the export behavior depends on the slab geometry (rectangular vs. non-rectangular) and the selected IFC schema version (IFC2x3 vs. IFC4).

Please ensure you are using appropriate Project Parameters for weight-related quantities and selecting the IFC4 Reference View for `Depth` and `Length` exports if needed.

Thank you for your patience and detailed reporting!

<img width="957" height="1710" alt="Image" src="https://github.com/user-attachments/assets/5da0cf45-14cc-4d38-937d-ae58f138be5a" />
